commands/command_utils.py

```diff
@@ -71,15 +99,33 @@ def get_upgrade_flavour():
     return LEAPP_UPGRADE_FLAVOUR_DEFAULT
 
 
-def _retrieve_os_release_contents(_os_release_path='/etc/os-release'):
+@functools.cache
```

Comment:
@MichalHe `functools.cache` is defined since Python 3.9, but the code is executed with Python 3.6 as well:
* https://docs.python.org/3/library/functools.html#functools.cache

Is there a high enough number of calls to have a benefit from using the cache wrapper at all?

Reply:
Thank you for noticing, I have dropped the use of `functools`.

> Is there a high enough number of calls to have a benefit from using the cache wrapper at all?

Since it prevents us from accessing the filesystem (systemcalls are usually expensive), then the cost of the wrapper is small. I somehow think that the performance of the tool (how fast can I iterate) is something that I don' enjoy about the development experience, so I try to take into consideration also that aspect of the code.